### ISSUE TITLE
Changelog sections are not generated in the order, they are specified in the list in changelog-types

### ISSUE THREAD

**Issue by runenielsen** [google-github-actions/release-please-action#760]
### TL;DR

Changelog sections are not generated in the order, they are specified in the list in changelog-types in the GitHub actions YAML file. It seems some sort of pseudo-random order is used.

### Expected behavior

That generated changelog sections for a release are in the same order, that they are specified in the GitHub actions YAML file.

### Observed behavior

I don't have any control over the order of generated sections in the CHANGELOG.md file. For the specified Action YAML below, I would for instance expect the "Bug Fixes" section to be placed before the "Features" section in the generated changelog for a release, but in reality "Features" is placed before "Bug Fixes".

### Action YAML

```yaml
name: Release Please

on:
  workflow_call:
    inputs:
      package-name:
        required: true
        type: string
      path:
        required: true
        type: string
      release-type:
        required: true
        type: string
      extra-files:
        required: false
        type: string
      changelog-path:
        required: false
        type: string
    outputs:
      released:
        value: ${{ jobs.release-please.outputs.released }}
      version:
        value: ${{ jobs.release-please.outputs.version }}

jobs:
  release-please:
    outputs:
      released: ${{ steps.release.outputs[format('{0}--release_created', inputs.path)] }}
      version: ${{ steps.release.outputs[format('{0}--release_created', inputs.path)] && format('{0}.{1}.{2}', steps.release.outputs[format('{0}--major', inputs.path)], steps.release.outputs[format('{0}--minor', inputs.path)], steps.release.outputs[format('{0}--patch', inputs.path)]) || '' }}
    runs-on: ubuntu-latest
    steps:
      - name: Release please
        uses: google-github-actions/release-please-action@v3
        id: release
        with:
          monorepo-tags: true
          package-name: ${{ inputs.package-name }}
          path: ${{ inputs.path }}
          release-type: ${{ inputs.release-type }}
          extra-files: |
            ${{ inputs.extra-files }}
          changelog-path: ${{ inputs.changelog-path }}
          changelog-types: >
            [
              {"type":"fix","section":"Bug Fixes"},
              {"type":"feat","section":"Features"}
            ]
```

**Comment by elijaholmos** [comment]
Encountered this issue as well and was surprised that it hadn't been addressed already. Seems intuitive that the order of `changelog-sections` in the config should be the order of the changelog sections in the release notes.

**Comment by chingor13** [comment]
Apologies for the late reply.

We highly recommend switching configuration to the release-please-config.json.

If this is still an issue, then it's in the upstream release-please codebase. Please reopen the issue there.
